### PR TITLE
refactor: normalize join_path path segments

### PR DESCRIPTION
**Pull request by AlexanderLanin**
## What this achieves

Adds focused unit coverage for the Starlark `join_path` helper and documents its path-normalization contract.

## Why this is needed

The helper now handles package-root paths represented by `.` and removes trailing slashes. These edge cases affect generated documentation paths and are easy to regress without direct coverage.

## Changes

- Normalize trailing slashes and `.` path segments in `join_path`.
- Add compact Starlark unit cases under `bzl/tests`.
- Add `bazel_skylib` as a development-only dependency for the unit-test framework.

### CODE REVIEW

**Review by a-zw — APPROVED** [review]
LGTM

File: bzl/tests/basics_test.bzl [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    """Check path joining and normalization at the Starlark level."""
+    env = unittest.begin(ctx)
+    # Each tuple contains ``prefix``, ``rest``, and the expected result.
+    cases = [
+        ("docs", "index", "docs/index"),
+        ("", "docs", "docs"),
+        (".", "docs", "docs"),
+        ("docs", "", "docs"),
+        ("package/docs", ".", "package/docs"),
+        ("package/", "docs/", "package/docs"),
```
Comment (MaximilianSoerenPollak) [inline bzl/tests/basics_test.bzl:28]: Should maybe also try to test `None` ?

**Review by MaximilianSoerenPollak — APPROVED** [review]
Looks good, maybe can in the future add further tests.

**Comment by AlexanderLanin** [comment]
yeah, good enough for now. not even sure how this should behave with None.
